dulwich/porcelain.py

```diff
@@ -426,71 +439,75 @@ def remove(repo=".", paths=None, cached=False):
 rm = remove
 
 
-def commit_decode(commit, contents, default_encoding=DEFAULT_ENCODING):
+def commit_decode(commit: Commit, contents: bytes, default_encoding: str=DEFAULT_ENCODING) -> str:
     if commit.encoding is not None:
         return contents.decode(commit.encoding, "replace")
     return contents.decode(default_encoding, "replace")
 
 
-def print_commit(commit, decode, outstream=sys.stdout):
+def print_commit(commit: Commit, decode: Callable[[bytes], str],
+                 out_stream=default_bytes_out_stream):
     """Write a human-readable commit log entry.
 
     :param commit: A `Commit` object
-    :param outstream: A stream file to write to
+    :param decode: Function for decoding bytes to unicode string
+    :param out_stream: A stream file to write to
     """
-    outstream.write("-" * 50 + "\n")
-    outstream.write("commit: " + commit.id.decode('ascii') + "\n")
+    out_stream.write("-" * 50 + "\n")
+    out_stream.write("commit: " + commit.id.decode('ascii') + "\n")
     if len(commit.parents) > 1:
-        outstream.write(
+        out_stream.write(
             "merge: " +
             "...".join([c.decode('ascii') for c in commit.parents[1:]]) + "\n")
-    outstream.write("Author: " + decode(commit.author) + "\n")
+    out_stream.write("Author: " + decode(commit.author) + "\n")
     if commit.author != commit.committer:
-        outstream.write("Committer: " + decode(commit.committer) + "\n")
+        out_stream.write("Committer: " + decode(commit.committer) + "\n")
 
     time_tuple = time.gmtime(commit.author_time + commit.author_timezone)
     time_str = time.strftime("%a %b %d %Y %H:%M:%S", time_tuple)
     timezone_str = format_timezone(commit.author_timezone).decode('ascii')
-    outstream.write("Date:   " + time_str + " " + timezone_str + "\n")
-    outstream.write("\n")
-    outstream.write(decode(commit.message) + "\n")
-    outstream.write("\n")
+    out_stream.write("Date:   " + time_str + " " + timezone_str + "\n")
+    out_stream.write("\n")
+    out_stream.write(decode(commit.message) + "\n")
+    out_stream.write("\n")
 
 
-def print_tag(tag, decode, outstream=sys.stdout):
+def print_tag(tag: Tag, decode: Callable[[bytes], str], out_stream=default_bytes_out_stream):
```

Comment:
The use of sys.stdout is intentional here, since defaults_bytes_out_stream doesn't support writing native strings on Python 3.